use C10_REGISTER for GELU op

hl475 opened the pull request:
Differential Revision: D14864737

houseroad reviewed (changes requested):
Please add the test here: https://github.com/pytorch/pytorch/blob/master/caffe2/python/operator_test/torch_integration_test.py

houseroad commented on caffe2/operators/gelu_op.cc:
```diff
@@ -130,3 +122,12 @@ class GetGeluGradient : public GradientMakerBase {
+    Gelu,
+    (std::vector<c10::Argument>{
+        c10::Argument("input"),
+        c10::Argument("fast_gelu", BoolType::get()),
```
set default value?

houseroad commented on caffe2/operators/gelu_op.cc:
```diff
@@ -7,6 +7,8 @@
 #include <string>
 #include <vector>
 
+#include <c10/core/Tensor.h>
```
why include Tensor.h?

houseroad commented on caffe2/python/operator_test/torch_integration_test.py:
```diff
@@ -447,6 +448,26 @@ def test_roi_align_cpu(self):
+        rtol = 1e-3 if fast_gelu else 1e-4
+        atol = 1e-5
+        torch.testing.assert_allclose(
+            expected_output, actual_output, rtol=rtol, atol=atol)
```
actual_output.cpu()

houseroad reviewed (commented):
Please address inline comments

hl475 commented:
@pytorchbot retest this please

houseroad reviewed (approved):
Looks good.

houseroad commented:
Let's wait a bit for the IC signals

hl475 committed:
use C10_REGISTER for GELU op

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/19090

Reviewed By: BIT-silence

Differential Revision: D14864737

fbshipit-source-id: d0298a0439388fbaed62cb27767a6e29f7a64e73
